libs/langchain-groq/src/chat_models.ts

```diff
@@ -0,0 +1,349 @@
+  }
+}
+
+function convertMessagesToGroqParams(
+  messages: BaseMessage[]
+): Array<ChatCompletion.Choice.Message> {
+  return messages.map((message) => {
+    if (typeof message.content !== "string") {
+      throw new Error("Non string message content not supported");
+    }
+    return {
+      role: messageToGroqRole(message),
+      content: message.content,
+      name: message.name,
+      function_call: message.additional_kwargs.function_call,
+    };
+  });
+}
+
+function groqResponseToChatMessage(
+  message: ChatCompletion.Choice.Message
+): BaseMessage {
+  switch (message.role) {
+    case "assistant":
+      return new AIMessage(message.content || "");
+    default:
+      return new ChatMessage(message.content || "", message.role ?? "unknown");
+  }
+}
+
+function _convertDeltaToMessageChunk(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  delta: Record<string, any>
+) {
+  const { role } = delta;
+  const content = delta.content ?? "";
+  let additional_kwargs;
+  if (delta.function_call) {
+    additional_kwargs = {
+      function_call: delta.function_call,
+    };
+  } else {
+    additional_kwargs = {};
+  }
+  if (role === "user") {
+    return new HumanMessageChunk({ content });
+  } else if (role === "assistant") {
+    return new AIMessageChunk({ content, additional_kwargs });
+  } else if (role === "system") {
+    return new SystemMessageChunk({ content });
+  } else {
+    return new ChatMessageChunk({ content, role });
+  }
+}
+
+/**
+ * Wrapper around Groq API for large language models fine-tuned for chat
+ *
+ * Groq API is compatible to the OpenAI API with some limitations. View the
+ * full API ref at:
+ * @link {https://docs.api.groq.com/md/openai.oas.html}
+ *
+ * To use, you should have the `GROQ_API_KEY` environment variable set.
+ * @example
+ * ```typescript
+ * const model = new ChatGroq({
+ *   temperature: 0.9,
+ *   apiKey: process.env.GROQ_API_KEY,
+ * });
+ *
+ * const response = await model.invoke([new HumanMessage("Hello there!")]);
+ * console.log(response);
+ * ```
+ */
+export class ChatGroq extends BaseChatModel<ChatGroqCallOptions> {
+  client: Groq;
+
+  modelName = "llama2-70b-4096";
+
+  temperature = 0.7;
+
+  static lc_name() {
+    return "ChatGroq";
+  }
+
+  _llmType() {
+    return "groq";
+  }
+
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {
+      apiKey: "GROQ_API_KEY",
+    };
+  }
+
+  lc_serializable = true;
+
+  constructor(fields?: ChatGroqInput) {
+    super(fields ?? {});
+
+    const apiKey = fields?.apiKey || getEnvironmentVariable("GROQ_API_KEY");
+    if (!apiKey) {
+      throw new Error(
+        `Groq API key not found. Please set the GROQ_API_KEY environment variable or provide the key into "apiKey"`
+      );
+    }
+
+    this.client = new Groq({
+      apiKey,
+    });
+    this.temperature = fields?.temperature ?? this.temperature;
+    this.modelName = fields?.modelName ?? this.modelName;
```

Comment:
can we pass the `stream` here?
`this.stream = fields?.stream ?? this.stream;`

Reply:
Hey, the recommended way of streaming is via the `.stream` method. Although this is a good catch, we should be passing the `streaming` class property here. Will add it today

Reply:
are we deprecating the `streaming` property as well (for other chat models)?

Reply:
@bracesproul we should add this - there are niche but legit use cases for invoke with aggregating token information.

Reply:
Will be in soon! https://github.com/langchain-ai/langchainjs/pull/4573/files